Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+struct SlideOverUXConstants {
+    static let ETPMenuHeightForGlobalOn: CGFloat = 385
+    static let ETPMenuHeightForGlobalOff: CGFloat = 285
+    static let ETPMenuCornerRadius: CGFloat = 8
+}
+
+class SlideOverPresentationController: UIPresentationController {
+
+    let blurEffectView: UIVisualEffectView!
+    var tapGestureRecognizer: UITapGestureRecognizer = UITapGestureRecognizer()
+    var globalETPStatus: Bool
+
+    init(presentedViewController: UIViewController, presenting presentingViewController: UIViewController?, withGlobalETPStatus status: Bool) {
+        self.globalETPStatus = status
+        let blurEffect = UIBlurEffect(style: .dark)
+        blurEffectView = UIVisualEffectView(effect: blurEffect)
+        super.init(presentedViewController: presentedViewController, presenting: presentingViewController)
+        tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissController))
+        blurEffectView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        self.blurEffectView.isUserInteractionEnabled = true
+        self.blurEffectView.addGestureRecognizer(tapGestureRecognizer)
+    }
+
+    override var frameOfPresentedViewInContainerView: CGRect {
+        var menuHeight: CGFloat
+        if globalETPStatus {
+            menuHeight = SlideOverUXConstants.ETPMenuHeightForGlobalOn
+        } else {
+            menuHeight = SlideOverUXConstants.ETPMenuHeightForGlobalOff
+        }
+
+        let yPosition = self.containerView!.frame.height - menuHeight
+        var xPosition: CGFloat = 0
+        var width: CGFloat = 0
+        if UIWindow.isLandscape {
+            width = 600
+            xPosition = self.containerView!.frame.width/2 - (width/2)
+        } else {
+            width = self.containerView!.frame.width
+        }
+        return CGRect(origin: CGPoint(x: xPosition, y: yPosition),
+                      size: CGSize(width: width, height: menuHeight))
+    }
+
+    override func presentationTransitionWillBegin() {
+        self.blurEffectView.alpha = 0
+        self.containerView?.addSubview(blurEffectView)
+        self.presentedViewController.transitionCoordinator?.animate(alongsideTransition: { (UIViewControllerTransitionCoordinatorContext) in
+            self.blurEffectView.alpha = 0.1
+        }, completion: { (UIViewControllerTransitionCoordinatorContext) in })
+    }
```

Comment:
nit: we may not need an explicit reference to `self` for these vars. And inside the closure, would we need to weakify `self`?